benchmark/utils/nodeRedUtils.ts

```diff
@@ -0,0 +1,144 @@
+    return false;
+  }
+}
+
+export async function executeEndpointFlow(endpointUrl: string, msg: any) {
+  
+  console.log(`Executing flow at endpoint: http://localhost:${NODE_RED_PORT}${endpointUrl}`);
+  
+  try {
+    const authHeader = generateAuthHeader(NODE_RED_USERNAME, NODE_RED_PASSWORD);
+    const headers: { [key: string]: string } = {
+      'Content-Type': 'application/json',
+      ...(authHeader ? { 'Authorization': authHeader } : {}),
+    };
+
+    const response = await fetch(
+      `http://localhost:${NODE_RED_PORT}${endpointUrl}`, {
+      method: 'POST',
+      headers: headers,
+      body: JSON.stringify(msg)
+    });
+
+    if (!response.ok) {
+      throw new Error(`HTTP error! status: ${response.status}`);
+    }
+
+    const responseData = await response.json();
+    console.log('Endpoint flow executed successfully.');
+
+    return responseData;
+
+  } catch (error) {
+    console.error('Error executing endpoint flow:', error);
+    throw error;
+  }
+}
+
+export async function updateFlow(flowId: string, nodes: any): Promise<boolean> {
+  console.log(`Attempting to update Node-RED flow with ID: ${flowId}`);
+
+  const authHeader = generateAuthHeader(NODE_RED_USERNAME, NODE_RED_PASSWORD);
+  const headers = {
+    'Content-Type': 'application/json',
+    ...(authHeader ? { 'Authorization': authHeader } : {}),
+  };
+
+  try {
+    const response = await fetch(`http://localhost:${NODE_RED_PORT}/flow/${flowId}`, {
+      method: 'PUT',
+      headers: headers,
+      body: JSON.stringify(nodes),
+    });
+
+    if (response.ok) {
+      console.log(`Flow with ID "${flowId}" updated successfully.`);
+      return true;
+    } else {
+      const errorText = await response.text();
+      console.error(`Failed to update flow "${flowId}". Status: ${response.status}, Response: ${errorText}`);
+      return false;
+    }
+  } catch (error) {
+    console.error('Error updating Node-RED flow:', error);
+    return false;
+  }
+}
+
+export async function deleteFlow(flowId: string): Promise<boolean> {
+  console.log(`Attempting to delete Node-RED flow: ${flowId}`);
+  const authHeader = generateAuthHeader(NODE_RED_USERNAME, NODE_RED_PASSWORD);
+  const headers = {
+    ...(authHeader ? { 'Authorization': authHeader } : {}),
+  };
+
+  try {
+    const response = await fetch(`http://localhost:${NODE_RED_PORT}/flow/${flowId}`, {
+      method: 'DELETE',
+      headers: headers,
+    });
+
+    if (response.ok) {
+      console.log(`Flow "${flowId}" deleted successfully.`);
+      return true;
+
+    } else if (response.status === 404) {
+      console.warn(`Flow "${flowId}" not found.`);
+      return true;
+      
+    } else {
+      const errorText = await response.text();
+      console.error(`Failed to delete flow "${flowId}". Status: ${response.status}, Response: ${errorText}`);
+      return false;
+    }
+  } catch (error) {
+    console.error('Error deleting Node-RED flow:', error);
+    return false;
+  }
+}
```

Comment:
Enable in VSCode the setting that makes all files end with a newline on save, so GitHub doesn't complain. This is important for interoperability reasons between operating systems.

```suggestion
}

```

The `sampleFlows.ts` file is also affected, update it as well please.

Reply:
Yeah this is a minor bug of the "npm run format" command, fixed too